[0.11.0] httpjson: panic: runtime error: invalid memory address or nil pointer dereference

After updating to version 0.11.0 telegraf started crashing at every collection cycle on one of 7 nodes.

Node used httpjson with following configuration:
```
   [[inputs.httpjson]]
        name='service_activeusers'
        method='GET'
        servers=['http://172.17.0.1:9030/active']
```

curl call works as expected:
```
[root@server telegraf]# curl http://172.17.0.1:9030/active
{"count":0}
```

```
[root@server telegraf]# telegraf -config /etc/telegraf/telegraf.conf -config-directory /etc/telegraf/telegraf.d
2016/03/16 15:23:45 Starting Telegraf (version 0.11.0)
2016/03/16 15:23:45 Loaded outputs: influxdb
2016/03/16 15:23:45 Loaded inputs: httpjson
2016/03/16 15:23:45 Tags enabled: host=server.example.org
2016/03/16 15:23:45 Agent Config: Interval:10s, Debug:false, Quiet:false, Hostname:"server.example.org", Flush Interval:10s
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x10 pc=0x6f80ad]

goroutine 31 [running]:
net/http.(*Client).doFollowingRedirects(0x0, 0xc820094380, 0x128b2a8, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:313 +0x6d
net/http.(*Client).Do(0x0, 0xc820094380, 0x4ccc78, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:175 +0x9c
github.com/influxdata/telegraf/plugins/inputs/httpjson.RealHTTPClient.MakeRequest(0x0, 0xc820094380, 0x17b3b20, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/httpjson/httpjson.go:59 +0x37
github.com/influxdata/telegraf/plugins/inputs/httpjson.(*HttpJson).sendRequest(0xc82016ebb0, 0xc8200ad1e0, 0x1d, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/httpjson/httpjson.go:263 +0x877
github.com/influxdata/telegraf/plugins/inputs/httpjson.(*HttpJson).gatherServer(0xc82016ebb0, 0x7ff3f61ada18, 0xc82016dd00, 0xc8200ad1e0, 0x1d, 0x0, 0x0)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/httpjson/httpjson.go:178 +0x5f
github.com/influxdata/telegraf/plugins/inputs/httpjson.(*HttpJson).Gather.func1(0xc8200ab030, 0xc82016ebb0, 0x7ff3f61ada18, 0xc82016dd00, 0xc820016de0, 0xc8200ad1e0, 0x1d)
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/httpjson/httpjson.go:145 +0x7e
created by github.com/influxdata/telegraf/plugins/inputs/httpjson.(*HttpJson).Gather
	/home/ubuntu/telegraf-build/src/github.com/influxdata/telegraf/plugins/inputs/httpjson/httpjson.go:148 +0x3ab
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,3 +1,11 @@
+## v0.11.1 [unreleased]
+
+### Features
+
+### Bugfixes
+- [#852](https://github.com/influxdata/telegraf/issues/852): Windows zip package fix
+- [#859](https://github.com/influxdata/telegraf/issues/859): httpjson plugin panic
+
 ## v0.11.0 [2016-03-15]
 
 ### Release Notes

--- a/plugins/inputs/httpjson/httpjson.go
+++ b/plugins/inputs/httpjson/httpjson.go
@@ -55,15 +55,15 @@ type RealHTTPClient struct {
 	client *http.Client
 }
 
-func (c RealHTTPClient) MakeRequest(req *http.Request) (*http.Response, error) {
+func (c *RealHTTPClient) MakeRequest(req *http.Request) (*http.Response, error) {
 	return c.client.Do(req)
 }
 
-func (c RealHTTPClient) SetHTTPClient(client *http.Client) {
+func (c *RealHTTPClient) SetHTTPClient(client *http.Client) {
 	c.client = client
 }
 
-func (c RealHTTPClient) HTTPClient() *http.Client {
+func (c *RealHTTPClient) HTTPClient() *http.Client {
 	return c.client
 }
 
@@ -289,6 +289,8 @@ func (h *HttpJson) sendRequest(serverURL string) (string, float64, error) {
 
 func init() {
 	inputs.Add("httpjson", func() telegraf.Input {
-		return &HttpJson{client: RealHTTPClient{}}
+		return &HttpJson{
+			client: &RealHTTPClient{},
+		}
 	})
 }

--- a/plugins/inputs/httpjson/httpjson_test.go
+++ b/plugins/inputs/httpjson/httpjson_test.go
@@ -125,7 +125,7 @@ type mockHTTPClient struct {
 // Mock implementation of MakeRequest. Usually returns an http.Response with
 // hard-coded responseBody and statusCode. However, if the request uses a
 // nonstandard method, it uses status code 405 (method not allowed)
-func (c mockHTTPClient) MakeRequest(req *http.Request) (*http.Response, error) {
+func (c *mockHTTPClient) MakeRequest(req *http.Request) (*http.Response, error) {
 	resp := http.Response{}
 	resp.StatusCode = c.statusCode
 
@@ -147,10 +147,10 @@ func (c mockHTTPClient) MakeRequest(req *http.Request) (*http.Response, error) {
 	return &resp, nil
 }
 
-func (c mockHTTPClient) SetHTTPClient(_ *http.Client) {
+func (c *mockHTTPClient) SetHTTPClient(_ *http.Client) {
 }
 
-func (c mockHTTPClient) HTTPClient() *http.Client {
+func (c *mockHTTPClient) HTTPClient() *http.Client {
 	return nil
 }
 
@@ -164,7 +164,7 @@ func (c mockHTTPClient) HTTPClient() *http.Client {
 func genMockHttpJson(response string, statusCode int) []*HttpJson {
 	return []*HttpJson{
 		&HttpJson{
-			client: mockHTTPClient{responseBody: response, statusCode: statusCode},
+			client: &mockHTTPClient{responseBody: response, statusCode: statusCode},
 			Servers: []string{
 				"http://server1.example.com/metrics/",
 				"http://server2.example.com/metrics/",
@@ -181,7 +181,7 @@ func genMockHttpJson(response string, statusCode int) []*HttpJson {
 			},
 		},
 		&HttpJson{
-			client: mockHTTPClient{responseBody: response, statusCode: statusCode},
+			client: &mockHTTPClient{responseBody: response, statusCode: statusCode},
 			Servers: []string{
 				"http://server3.example.com/metrics/",
 				"http://server4.example.com/metrics/",
@@ -241,7 +241,7 @@ func TestHttpJsonGET_URL(t *testing.T) {
 		Servers: []string{ts.URL + "?api_key=mykey"},
 		Name:    "",
 		Method:  "GET",
-		client:  RealHTTPClient{client: &http.Client{}},
+		client:  &RealHTTPClient{client: &http.Client{}},
 	}
 
 	var acc testutil.Accumulator
@@ -314,7 +314,7 @@ func TestHttpJsonGET(t *testing.T) {
 		Name:       "",
 		Method:     "GET",
 		Parameters: params,
-		client:     RealHTTPClient{client: &http.Client{}},
+		client:     &RealHTTPClient{client: &http.Client{}},
 	}
 
 	var acc testutil.Accumulator
@@ -388,7 +388,7 @@ func TestHttpJsonPOST(t *testing.T) {
 		Name:       "",
 		Method:     "POST",
 		Parameters: params,
-		client:     RealHTTPClient{client: &http.Client{}},
+		client:     &RealHTTPClient{client: &http.Client{}},
 	}
 
 	var acc testutil.Accumulator

